### PR TITLE
Sensirion Sht4x Support 

### PR DESCRIPTION
**Pull request by trichner**
Support for the SHT4x family of humidity & temperature sensors from Sensirion.

I've tried to keep the interface aligned with the existing SHT3x driver where it made sense but kept it
simple.

### CODE REVIEW

**Comment by deadprogram** [comment]
Hello @trichner thanks for the new device support.

Can you please also add it to the smoketests here? https://github.com/tinygo-org/drivers/blob/dev/smoketest.sh

**Comment by trichner** [comment]
> Hello @trichner thanks for the new device support.
> 
> Can you please also add it to the smoketests here? https://github.com/tinygo-org/drivers/blob/dev/smoketest.sh

Done!
Though running them fails on some other test locally:
```
> make test
...
src/machine/board_bluepill.go:6:2: package device/stm32 is not in GOROOT (<redacted>/.cache/tinygo/goroot-3db76b2bbe6512878c9fc4ff18b9674eba0696a404ae4e2915298e4f7a087fad/src/device/stm32)
```

File: sht4x/sht4x.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+// Device represents a SHT4x sensor
+type Device struct {
+	bus     drivers.I2C
+	address uint8
+}
+
+// New creates a new SHT4x connection. The I2C bus must already be
+// configured.
+func New(bus drivers.I2C, addr uint8) Device {
```
Comment (deadprogram) [inline sht4x/sht4x.go]: I would be better to follow the pattern used by most drivers here, which is 

```go
type Device struct {
	bus     drivers.I2C
	Address uint8
}

func New(bus drivers.I2C) Device {
        return Device{
		bus:     bus,
		Address: DefaultAddress,
	}
}
```

If the dev wants to set the address to something else, it is then a matter of

```go
d := sht4x.New(machine.I2C0)
d.Address = 0x45
```

Comment (trichner) [reply sht4x/sht4x.go]: Done!

**Comment by deadprogram** [comment]
Thank you for making the requested changes @trichner now going to squash/merge.

**Pull request merged**